site/_collections/feeds.js

```diff
@@ -106,6 +109,21 @@ module.exports = collection => {
         tagsFeeds[tag].items.push(post);
       }
     }
+
+    const authors = [post.data.authors ?? []].flat();
+    for (const author of authors) {
+      // If author feed does not exist in feeds yet, create FeedsCollectionItem
+      if (!authorsFeeds[author]) {
```

Comment:
This condition is only necessary because `authors` might contain duplicate authors, no? You could instead think about converting `authors` to a `Set` which would remove duplicates and save unnecessary loops.

Reply:
Yes, this condition is for checking the author key whether exists in the `authorsFeeds` to avoid adding duplicate authors.

However, I don't think we can remove this condition. According to the current logic, we loop over posts, then list all authors for each post, and loop over the authors to add the current post into the author key in the  `authorsFeeds`. I think we need to keep `!authorsFeeds[author]` condition to check whether we have this author in the `authorsFeeds`. If so just add a current post into the `item`. If not, create a new author key in `authorsFeeds` which is the same code as in that condition. 

The result of the authorsFeeds will be like this: 
```
{
  permalink: '/authors/paulkinlan/feed.xml',
  title: 'Paul Kinlan',
  url: '/authors/paulkinlan',
  items: [
    {
      template: [Template],
      inputPath: './site/en/blog/insider-dec-22/index.md',
      fileSlug: 'insider-dec-22',
      filePathStem: '/en/blog/insider-dec-22/index',
      data: [Object],
      date: 2022-12-15T00:00:00.000Z,
      outputPath: 'dist/en/blog/insider-dec-22/index.html',
      url: '/en/blog/insider-dec-22/',
      checkTemplateContent: true,
      templateContent: [Getter/Setter]
    },
    {
      template: [Template],
      inputPath: './site/en/blog/insider-oct-2022/index.md',
      fileSlug: 'insider-oct-2022',
      filePathStem: '/en/blog/insider-oct-2022/index',
      data: [Object],
      date: 2022-10-05T00:00:00.000Z,
      outputPath: 'dist/en/blog/insider-oct-2022/index.html',
      url: '/en/blog/insider-oct-2022/',
      checkTemplateContent: true,
      templateContent: [Getter/Setter]
    },
    ...
  ]
}
```

Please feel free to let me know if I am wrong about this 😊

Reply:
Sorry Maya, I should have expanded the code a little more for full context. I had the wrong assumptions, what you originally had was perfectly fine 🙂